pgmpy/readwrite/BIF.py

```diff
@@ -0,0 +1,312 @@
+
+        if path:
+            self.network = open(path, 'r').read()                                               
+
+        elif string:
+            self.network = string        
+
+        else:
+            raise ValueError("Must specify either path or string")
+
+        if '"' in self.network:
+            """
+            Replacing quotes by spaces to remove case sensitivity like:
+            "Dog-Problem" and Dog-problem
+            or "true""false" and "true" "false" and true false
+            """
+            self.network = self.network.replace('"', ' ')
+
+        self.network = multi_space.setParseAction(lambda t:' ').transformString(self.network)   # replacing mulitple spaces or tabs by one space
+
+        if '/*' or '//' in self.network:
+            self.network = cppStyleComment.suppress().transformString(self.network)             # removing comments from the file
+
+        self.get_network_name()
+        self.get_variables_info()
+        self.get_cpd()
+        self.get_edges()
+        self.model = self.get_model()
+
+    def get_network_name(self):
+
+        """
+        Retruns the name of the network
+
+        network attributes are of the format
+            network <name_of_network> {
+                attirbutes
+            }
+        Example of network attribute
+        ------------------------------
+        network "Dog-Problem" { //5 variables and 5 probability distributions
+                property "credal-set constant-density-bounded 1.1" ;
+        }
+
+        Sample run
+        ---------------
+        >>> reader = BIF.BifReader("bif_test.bif")
+        >>> reader.network_name()
+        'Dog-Problem'
+        """
+        network_attribute = Suppress('network') + Word(alphanums+'_'+'-') + '{'                            # Creating a network attribute 
+        temp = network_attribute.searchString(self.network)
+        self.network_name = temp[0][0]
+        return self.network_name
+
+    def get_variables_info(self):
+
+        """
+        Functions gets all type of variable information
+
+        variable block is of format
+
+        variable <name_of_variable> {
+            attribute1
+            attribute2
+            ...
+        }
+        Example of variable attribute
+        ----------
+        variable  "light-on" { //2 values
+	        type discrete[2] {  "true"  "false" };
+            property "position = (218, 195)" ;
+        }
+        """
+        variable_block_starts = [x.start() for x in re.finditer('variable', self.network)]  # Finding the beginning of variable block
+        variable_block = []
+
+        for i in variable_block_starts:
+            variable_block_end = self.network.find('}\n', i)                                # Finding the end of variable block
+            variable_block.append( self.network[ i : variable_block_end ] )                 # Appending the complete variable block in a list
```

Comment:
@khalibartan Add comments only if you are doing something complex and the comment would actually help in understanding it quickly.